src/sqlancer/stonedb/StoneDBSchema.java

```diff
@@ -0,0 +1,382 @@
+            this.isPrimaryKey = isPrimaryKey;
+            this.isNullable = isNullable;
+            this.precision = precision;
+        }
+
+        public boolean isPrimaryKey() {
+            return isPrimaryKey;
+        }
+
+        public boolean isNullable() {
+            return isNullable;
+        }
+
+        public int getPrecision() {
+            return precision;
+        }
+
+    }
+
+    public StoneDBSchema(List<StoneDBTable> databaseTables) {
+        super(databaseTables);
+    }
+
+    public static class StoneDBCompositeDataType {
+        private final StoneDBDataType dataType;
+        private final int size;
+
+        public StoneDBCompositeDataType(StoneDBDataType dataType, int size) {
+            this.dataType = dataType;
+            this.size = size;
+        }
+
+        public StoneDBCompositeDataType(StoneDBDataType dataType) {
+            this.dataType = dataType;
+            int size = -1;
+            switch (dataType) {
+            case TINYINT:
+                size = 1;
+                break;
+            case SMALLINT:
+                size = 2;
+                break;
+            case MEDIUMINT:
+                size = 3;
+                break;
+            case INT:
+                size = 4;
+                break;
+            case BIGINT:
+                size = 8;
+                break;
+            case FLOAT:
+                size = 4;
+                break;
+            case DOUBLE:
+                size = 8;
+                break;
+            case DECIMAL:
+                size = -2;
+                break;
+            case YEAR:
+                size = -2;
+                break;
+            case TIME:
+                size = -2;
+                break;
+            case DATE:
+                size = -2;
+                break;
+            case DATETIME:
+                size = -2;
+                break;
+            case TIMESTAMP:
+                size = -2;
+                break;
+            case CHAR:
+                size = -2;
+                break;
+            case VARCHAR:
+                size = -2;
+                break;
+            case TINYTEXT:
+                size = -2;
+                break;
+            case TEXT:
+                size = -2;
+                break;
+            case MEDIUMTEXT:
+                size = -2;
+                break;
+            case LONGTEXT:
+                size = -2;
+                break;
+            case BINARY:
+                size = -2;
+                break;
+            case VARBINARY:
+                size = -2;
+                break;
+            case TINYBLOB:
+                size = -2;
+                break;
+            case BLOB:
+                size = -2;
+                break;
+            case MEDIUMBLOB:
+                size = -2;
+                break;
+            case LONGBLOB:
+                size = -2;
+                break;
+            default:
+                size = -1;
```

Comment:
I guess this assignment doesn't make sense, since you throw an exception right after?

Reply:
oh yes, make sense, I just want to let `-1` represents the abnormal value here.